status/reporters.go

```diff
@@ -289,6 +328,43 @@ func (r *GitHubReporter) createCommitStatus(k8sClient client.Client, ctx context
 	return nil
 }
 
+func (r *GitHubReporter) createCommitStatusAdapterForSnapshot(snapshot *applicationapiv1alpha1.Snapshot, integrationTestStatusDetail gitops.IntegrationTestStatusDetail, owner, repo, sha string) (*github.CommitStatusAdapter, error) {
+	var (
+		state       string
+		description string
+	)
+
+	snapshotName := snapshot.Name
+	scenarioName := integrationTestStatusDetail.ScenarioName
+	statusContext := NamePrefix + " / " + snapshot.Name + " / " + scenarioName
+
+	switch integrationTestStatusDetail.Status {
+	case gitops.IntegrationTestStatusEnvironmentProvisionError:
+		state = "errorOccured"
+		description = "snapshot " + snapshotName + " experienced error when provisioning environment for integrationTestScenario " + scenarioName
+	case gitops.IntegrationTestStatusDeploymentError:
+		state = "errorOccured"
+		description = "snapshot " + snapshotName + " experienced error when deploying snapshotEnvironmentBinding for integrationTestScenario " + scenarioName
+	case gitops.IntegrationTestStatusTestPassed:
+		state = "success"
+		description = "snapshot " + snapshotName + " has passed integration test against integrationTestScenario " + scenarioName
+	case gitops.IntegrationTestStatusTestFail:
+		state = "failure"
+		description = "snapshot " + snapshotName + " has failed integration test against integrationTestScenario " + scenarioName
+	default:
+		return nil, fmt.Errorf("unknown status %s for integrationTestScenario %s/%s", integrationTestStatusDetail.Status, snapshot.Namespace, scenarioName)
+	}
```

Comment:
Why do I see the exactly same code twice as in previous function?
Can we exctract it into shared function?

Reply:
I feel the same, let me update it. Thanks

Reply:
fixed.